test/e2e/clusterctl_upgrade.go

```diff
@@ -447,6 +447,9 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 		})
 		Expect(workloadClusterTemplate).ToNot(BeNil(), "Failed to get the cluster template")
 
+		log.Logf("Applying the cluster template yaml to the cluster in dry-run")
```

Comment:
Let's add a short comment why we are doing this (ensuring the webhooks are all up & working)